packages/firebase_ml_vision/example/lib/main.dart

```diff
@@ -2,32 +2,151 @@
+              info.image.width.toDouble(),
+              info.image.height.toDouble(),
+            )));
+    return completer.future;
+  }
+
+  Widget _buildImage() {
+    return new Container(
+      constraints: const BoxConstraints.expand(),
+      decoration: new BoxDecoration(
+        image: new DecorationImage(
+          image: Image.file(_imageFile).image,
+          fit: BoxFit.fill,
+        ),
+      ),
+      child: _imageSize == null || _textLocations == null
+          ? const Center(
+              child: const Text(
+              "Scanning...",
+              style: const TextStyle(
+                color: Colors.green,
+                fontSize: 30.0,
+              ),
+            ))
+          : new CustomPaint(
+              painter: new ScannedTextPainter(_imageSize, _textLocations),
+            ),
+    );
   }
 
   @override
   Widget build(BuildContext context) {
-    return new MaterialApp(
-      home: new Scaffold(
-        appBar: new AppBar(
-          title: const Text('Plugin example app'),
-        ),
-        body: const Center(
-          child: const Text('Hello, World!'),
-        ),
+    return new Scaffold(
+      appBar: new AppBar(
+        title: const Text('ML Vision Example'),
+      ),
+      body: _imageFile == null
+          ? const Center(child: const Text("No image selected."))
+          : _buildImage(),
+      floatingActionButton: new FloatingActionButton(
+        onPressed: () async {
+          await _getImage();
+          if (_imageFile != null) _scanImage();
+        },
+        tooltip: 'Pick Image',
+        child: const Icon(Icons.add_a_photo),
       ),
     );
   }
 }
+
+class ScannedTextPainter extends CustomPainter {
+  ScannedTextPainter(this.absoluteImageSize, this.textLocations);
+
+  final Size absoluteImageSize;
+  final List<TextBlock> textLocations;
+
+  @override
+  void paint(Canvas canvas, Size size) {
+    final double scaleX = size.width / absoluteImageSize.width;
+    final double scaleY = size.height / absoluteImageSize.height;
+
+    Rect scaleRect(TextContainer container) {
+      return new Rect.fromLTRB(
+        container.boundingBox.left * scaleX,
+        container.boundingBox.top * scaleY,
+        container.boundingBox.right * scaleX,
+        container.boundingBox.bottom * scaleY,
+      );
+    }
+
+    final Paint paint = new Paint()
+      ..style = PaintingStyle.stroke
+      ..strokeWidth = 2.0;
+
+    for (TextBlock block in textLocations) {
+      for (TextLine line in block.lines) {
+        for (TextElement element in line.elements) {
+          paint.color = Colors.green;
+          canvas.drawRect(scaleRect(element), paint);
+        }
+
+        paint.color = Colors.yellow;
+        canvas.drawRect(scaleRect(line), paint);
+      }
+
+      paint.color = Colors.red;
+      canvas.drawRect(scaleRect(block), paint);
+    }
+  }
+
+  @override
+  bool shouldRepaint(CustomPainter oldDelegate) => false;
```

Comment:
Why is this always false, shouldn't it repaint if the image or the identified text changes?